nilearn/decoding/tests/test_searchlight.py

```diff
@@ -62,3 +62,14 @@ def test_searchlight():
     sl.fit(data_img, cond)
     assert_equal(np.where(sl.scores_ == 1)[0].size, 33)
     assert_equal(sl.scores_[2, 2, 2], 1.)
+
+    # Regression test for #1216: if process_mask_img is not given, no masking is
+    # applied instead of using mask_img
+    mask = np.zeros((5, 5, 5), np.bool)
+    mask[0, 0, 0] = True
+    mask_img = nibabel.Nifti1Image(mask.astype(np.int), np.eye(4))
+
+    sl = searchlight.SearchLight(mask_img, radius=4,
+                                 n_jobs=n_jobs, scoring='accuracy', cv=cv)
+    sl.fit(data_img, cond)
+    assert_equal(np.where(sl.scores_ == 1)[0].size, 0)
```

Comment:
Sorry, I am missing the point. The scores are uniformly zero. Shouldn't we have a non-zero score at [0, 0, 0].
Maybe I missed the logic of this update.